shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz_code_blanks/presentation/StepQuizCodeBlanksReducer.kt

```diff
@@ -175,4 +163,39 @@ class StepQuizCodeBlanksReducer(
             }
         }
     }
+
+    private fun handleEnterButtonClicked(
+        state: State
+    ): StepQuizCodeBlanksReducerResult? {
+        if (state !is State.Content) {
+            return null
+        }
+
+        val activeCodeBlockIndex = state.activeCodeBlockIndex()
+
+        val actions = setOf(
+            InternalAction.LogAnalyticEvent(
+                StepQuizCodeBlanksClickedEnterHyperskillAnalyticEvent(
+                    route = stepRoute.analyticRoute,
+                    codeBlock = activeCodeBlockIndex?.let { state.codeBlocks[it] }
+                )
+            )
+        )
+
+        return if (activeCodeBlockIndex != null) {
+            val newCodeBlocks = state.codeBlocks.mutate {
+                set(activeCodeBlockIndex, copyCodeBlock(state.codeBlocks[activeCodeBlockIndex], isActive = false))
+                add(CodeBlock.Blank(isActive = true))
```

Comment:
New Blank will be added to the end of the list. As I understand from the task description, clicking the Enter button new Blank should be added just after the active code block, not in the end of the list.

Reply:
Thanks, you are right 👍